app/src/main/java/org/mozilla/focus/observer/LoadingObserver.java

```diff
@@ -0,0 +1,46 @@
+package org.mozilla.focus.observer;
+
+import android.arch.lifecycle.LifecycleOwner;
+import android.os.SystemClock;
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import org.mozilla.focus.architecture.NonNullObserver;
+import org.mozilla.focus.session.Session;
+import org.mozilla.focus.telemetry.TelemetryWrapper;
+
+public class LoadingObserver {
```

Comment:
\[PMD\] INFO: The class 'LoadingObserver' has a Cyclomatic Complexity of 6 \(Highest = 5\)\.